benchmark/primitives/primitives_test.go

```diff
@@ -188,6 +199,83 @@ func BenchmarkMutexWithoutDefer(b *testing.B) {
 	}
 }
 
+func BenchmarkAtomicAddInt64(b *testing.B) {
+	var c int64
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		atomic.AddInt64(&c, 1)
+	}
+	b.StopTimer()
+	if c != int64(b.N) {
+		b.Fatal("error")
+	}
+}
+
+func BenchmarkAtomicTimeValueStore(b *testing.B) {
+	var c atomic.Value
+	t := time.Now()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		c.Store(t)
+	}
+	b.StopTimer()
+}
+
+func atomicStore(a int) {
+	var wg sync.WaitGroup
+	var c atomic.Value
+	for i := 0; i < a; i++ {
+		wg.Add(1)
+		go func() {
+			c.Store(a)
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+}
+
+func mutexStore(a int) {
+	var wg sync.WaitGroup
+	mu := sync.Mutex{}
+	var c int
+	for i := 0; i < a; i++ {
+		wg.Add(1)
+		go func() {
+			mu.Lock()
+			c = a
+			mu.Unlock()
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+}
+
+func BenchmarkValueStoreWithContention(b *testing.B) {
+	benchmarks := []struct {
+		name string
+		n    int
+	}{
+		{"1", 1},
+		{"10", 10},
+		{"100", 100},
+		{"1000", 1000},
+		{"10000", 10000},
+		{"100000", 100000},
+	}
+	for _, bm := range benchmarks {
+		b.Run("Atomic/"+bm.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
```

Comment:
This doesn't seem right to me.  I think you want to push `b.N` into `atomicStore`/`mutexStore` and put that loop inside the goroutine.  Otherwise I'm not sure the contention will be as high because a lot more time would be spent spooling up and tearing down goroutines and in waitgroup accouting.

Reply:
Did the scaling look different with this change?  Can you include all the new numbers in the PR?